shell/platform/fuchsia/flutter/platform_view.cc

```diff
@@ -538,6 +539,44 @@ static trace_flow_id_t PointerTraceHACK(float fa, float fb) {
   return (((uint64_t)ia) << 32) | ib;
 }
 
+// For certain scenarios that must avoid floating-point drift, compute a
+// coordinate that falls within the logical view bounding box.
+std::array<float, 2> PlatformView::ClampToViewSpace(const float x,
+                                                    const float y) const {
+  if (!view_logical_size_.has_value() || !view_logical_origin_.has_value()) {
+    return {x, y};  // If we can't do anything, return the original values.
+  }
+
+  const auto origin = view_logical_origin_.value();
+  const auto size = view_logical_size_.value();
+  const float min_x = origin[0];
+  const float max_x = origin[0] + size[0];
+  const float min_y = origin[1];
+  const float max_y = origin[1] + size[1];
+  if (min_x <= x && x < max_x && min_y <= y && y < max_y) {
+    return {x, y};  // No clamping to perform.
+  }
+
+  // View boundary is [min_x, max_x) x [min_y, max_y). Note that min is
+  // inclusive, but max is exclusive - so we subtract epsilon.
+  const float epsilon = 0.0001f;
```

Comment:
std::numeric_limits<float>::epsilon and std::clamp are your friend here (and preferred)

Reply:
Nice, thanks.